src/Microsoft.Fx.Portability/ApiPortService.cs

```diff
@@ -156,29 +157,42 @@ public ApiPortService(string endpoint, ProductInformation info)
             return await _client.CallAsync<IEnumerable<ResultFormatInformation>>(HttpMethod.Get, Endpoints.ResultFormat);
         }
 
+        public async Task<ServiceResponse<ResultFormatInformation>> GetDefaultResultFormatAsync()
+        {
+            return await _client.CallAsync<ResultFormatInformation>(HttpMethod.Get, Endpoints.DefaultResultFormat);
+        }
+
         public void Dispose()
         {
             _client.Dispose();
         }
 
         private async Task<IEnumerable<ResultFormatInformation>> GetResultFormat(IEnumerable<string> format)
         {
-            var requestedFormats = new HashSet<string>(format, StringComparer.OrdinalIgnoreCase);
-            var formats = await GetResultFormatsAsync();
-            var formatInformation = formats.Response
-                .Where(r => requestedFormats.Contains(r.DisplayName))
-                .ToList();
-
-            var unknownFormats = requestedFormats
-                .Except(formatInformation.Select(f => f.DisplayName), StringComparer.OrdinalIgnoreCase)
-                .ToList();
-
-            if (unknownFormats.Any())
+            if (format == null || !format.Any())   //no "resultFormat" string option provider by user
             {
-                throw new UnknownReportFormatException(unknownFormats);
+                var defaultFormat = await GetDefaultResultFormatAsync();
+                return new[] { defaultFormat.Response };
+            }
+            else
+            {
+                var requestedFormats = new HashSet<string>(format, StringComparer.OrdinalIgnoreCase);
+                var formats = await GetResultFormatsAsync();
+                var formatInformation = formats.Response
+                    .Where(r => requestedFormats.Contains(r.DisplayName))
+                    .ToList();
+
+                var unknownFormats = requestedFormats
+                    .Except(formatInformation.Select(f => f.DisplayName), StringComparer.OrdinalIgnoreCase)
+                    .ToList();
```

Comment:
Remove `.ToList();` here and on line 183. We can lazily evaluate it rather than forcing the results to be evaluated at that time.